main.py

```diff
@@ -760,17 +757,18 @@ def handle_link_down(self, interface):
             link.deactivate()
             link.update_metadata('last_status_change', time.time())
             link.update_metadata('last_status_is_active', False)
-            self.notify_topology_update()
             self.notify_link_status_change(link, reason='link down')
+            self.notify_topology_update()
         if link and not link.is_active():
             with self._links_lock:
                 last_status = link.get_metadata('last_status_is_active')
                 if last_status:
                     link.update_metadata('last_status_is_active', False)
                     link.update_metadata('last_status_change', time.time())
-                    self.notify_topology_update()
                     self.notify_link_status_change(link, reason='link down')
+                    self.notify_topology_update()
         interface.deactivate()
+        self.notify_topology_update()
```

Comment:
Why notifying in lines 761 and 769 if line 771 will always be reached?

Reply:
I ended up following the approach of notifying as soon as any network changes take place to update the graph accordingly, but yeah, since `notify_topology_update` it report the entire graph, it's safe to get rid of them, and only notify at the end here. I'll remove them.

Reply:
done here https://github.com/kytos-ng/topology/pull/50/commits/19b28e70eca8883d14a5f92bae354daeeeca5395 @ajoaoff